lib/commands/general.js

```diff
@@ -231,31 +231,15 @@ commands.openNotifications = async function () {
 };
 
 commands.updateSettings = async function (settings) {
-  // we have some settings that are set on the settings object in the driver
-  // only, for example image finding settings. The uiauto2 server does not know
-  // what to do with them, so just set them on this driver's settings instance,
-  // and don't forward them to the server
-  let driverOnlySettings = {};
-  let serverSettings = {};
-  for (let [setting, value] of _.toPairs(settings)) {
-    if (BASEDRIVER_HANDLED_SETTINGS.includes(setting)) {
-      driverOnlySettings[setting] = value;
-    } else {
-      serverSettings[setting] = value;
-    }
-  }
-  if (!_.isEmpty(driverOnlySettings)) {
-    log.info(`Found some settings designed to be handled by BaseDriver: ` +
-             `${JSON.stringify(_.keys(driverOnlySettings))}. Not ` +
-             `sending these on to the UiAutomator2 server and instead ` +
-             `setting directly on the driver`);
-    await this.settings.update(driverOnlySettings);
-  }
-  if (!_.isEmpty(serverSettings)) {
-    log.info('Forwarding the following settings to the UiAutomator2 server: ' +
-             JSON.stringify(_.keys(serverSettings)));
-    await this.uiautomator2.jwproxy.command('/appium/settings', 'POST',
-      {settings: serverSettings});
+  // we have some settings that are set on the settings object in the driver only, for example
+  // settings used by plugins. The uiauto2 server does not know what to do with them, so make sure
+  // to set them on the driver settings object as well
+  await this.settings.update(settings);
+  try {
+    await this.uiautomator2.jwproxy.command('/appium/settings', 'POST', {settings});
```

Comment:
this does not look right to me. the server part must be updated instead

Reply:
Prepared https://github.com/appium/appium-uiautomator2-server/pull/457

Reply:
that PR will not address the need here. if we want to address this in the server it would need to not only throw no exceptions, but also store the value so it can be retrieved. in other words, settings can't be required to be known in advance.

this PR solves that issue by always putting each setting on the JS object.